contract/src/lib.rs

```diff
@@ -540,6 +563,136 @@ impl BtcLightClient {
         );
     }
 
+    #[cfg(feature = "zcash")]
+    fn zcash_check_pow_and_equihash(
+        &self,
+        block_header: &Header,
+        prev_block_header: &ExtendedHeader,
+    ) {
+        let config = self.get_config();
+
+        if let Some(pow_allow_min_difficulty_blocks_after_height) =
+            config.pow_allow_min_difficulty_blocks_after_height
+        {
+            // Special difficulty rule for testnet:
+            // If the new block's timestamp is more than 6 * block interval minutes
+            // then allow mining of a min-difficulty block.
+            if prev_block_header.block_height >= pow_allow_min_difficulty_blocks_after_height {
+                if i64::from(block_header.time)
+                    > i64::from(prev_block_header.block_header.time)
+                        + config.pow_target_spacing(prev_block_header.block_height + 1) * 6
+                {
+                    let expected_bits = config.proof_of_work_limit_bits;
+                    require!(
+                        expected_bits == block_header.bits,
```

Comment:
There are 4 places where the target should be verified, so we should just return it and verify it in one place